DeepProjection/tests/test_pdbGenerator.py

```diff
@@ -13,4 +13,182 @@ def test_getting_pdb(empty_pdbgenerator):
+    assert empty_pdbgenerator.download_pdb_in_range('AAA*', '1BB*', 'FAKEDIR') == False
+    assert empty_pdbgenerator.download_pdb_in_range('AAA', '1BB', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('AAA', '1BB', 'FAKEDIR') == False
+    assert empty_pdbgenerator.download_pdb_in_range('A*', '1*', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('A*', '1*', 'FAKEDIR') == False
+    assert empty_pdbgenerator.download_pdb_in_range('A', '1', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('A', '1', 'FAKEDIR') == False
+
+    # Testing Input Error #4: len(lower) or len(upper) is greater than 4; not proper PDB ID format.
+    assert empty_pdbgenerator.download_pdb_in_range('10000', '20000', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('10000', '20000', 'FAKEDIR') == False
+    assert empty_pdbgenerator.download_pdb_in_range('1AAAA', '2BBBB', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('1AAAA', '2BBBB', 'FAKEDIR') == False
+
+    assert empty_pdbgenerator.download_pdb_in_range('1000*', '2000*', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('1000*', '2000*', 'FAKEDIR') == False
+    assert empty_pdbgenerator.download_pdb_in_range('1AAA*', '2BBB*', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('1AAA*', '2BBB*', 'FAKEDIR') == False
+
+    # Testing Input Error #5: lower and upper contain upper case letters; just convert lower and upper to lower case.
+    # These download_pdb_in_range() calls should return True, as the code should convert all strings to lowercase.
+    assert empty_pdbgenerator.download_pdb_in_range('9Z00', '9Z01', None) == True
+    assert empty_pdbgenerator.download_pdb_in_range('9Z00', '9Z01', 'FAKEDIR') == True
+
+    assert empty_pdbgenerator.download_pdb_in_range('9Zz0', '9Zz1', None) == True
+    assert empty_pdbgenerator.download_pdb_in_range('9Zz0', '9Zz1', 'FAKEDIR') == True
+
+    assert empty_pdbgenerator.download_pdb_in_range('9AAY', '9AAZ', None) == True
+    assert empty_pdbgenerator.download_pdb_in_range('9AAY', '9AAZ', 'FAKEDIR') == True
+    
+    assert empty_pdbgenerator.download_pdb_in_range('9AaY', '9AaZ', None) == True
+    assert empty_pdbgenerator.download_pdb_in_range('9AaY', '9AaZ', 'FAKEDIR') == True
+
+    # Testing Input Error #6: The first character is '0'; not proper PDB format.
+    assert empty_pdbgenerator.download_pdb_in_range('0Z00', '0Z01', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('0Z00', '0Z01', 'FAKEDIR') == False
+
+    assert empty_pdbgenerator.download_pdb_in_range('7Zz0', '0Zz1', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('7Zz0', '0Zz1', 'FAKEDIR') == False
+
+    assert empty_pdbgenerator.download_pdb_in_range('0AAY', '7AAZ', None) == False
+    assert empty_pdbgenerator.download_pdb_in_range('0AAY', '7AAZ', 'FAKEDIR') == False
```

Comment:
there are too many tests here. Once you test for upper and lower cases, there's no need to keep on testing this
Same with 'FAKEDIR' test

let's only do 1 or 2 per input error type.